xdsl/dialects/memref.py

```diff
@@ -501,20 +502,21 @@ def get(
         sym_type: Attribute,
         initial_value: Attribute,
         sym_visibility: StringAttr = StringAttr("private"),
+        constant: bool | None = None,
         alignment: int | IntegerAttr[IntegerType] | None = None,
     ) -> Global:
         if isinstance(alignment, int):
             alignment = IntegerAttr.from_int_and_width(alignment, 64)
-
-        return Global.build(
-            properties={
-                "sym_name": sym_name,
-                "type": sym_type,
-                "initial_value": initial_value,
-                "sym_visibility": sym_visibility,
-                "alignment": alignment,
-            }
-        )
+        props: dict[str, Attribute | None] = {
+            "sym_name": sym_name,
+            "type": sym_type,
+            "initial_value": initial_value,
+            "sym_visibility": sym_visibility,
+            "alignment": alignment,
+        }
+        if constant is not None and constant:
+            props["constant"] = UnitAttr()
```

Comment:
It's probably worth changing the `constant` to be an optional UnitAttr, so this logic can be replaced by just `"constant": constant`, so the dictionary literal can move back into the `.build` helper